aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -258,19 +272,46 @@ Tensor _bincount_cuda_template(
     AT_ERROR("input and weights should have the same length");
   }
 
-  auto nbins = self.max().item<int64_t>() + 1L;
-  nbins = std::max(nbins, minlength);
+  const int64_t nbins = std::max(*self.max().cpu().data<input_t>() + 1L, minlength);
```

Comment:
This is probably fine because it compiled, but I wonder how c++ compiler treats this when input_t isn't integral

Reply:
It will throw an error, `TypeError: histc(): argument 'bins' must be int, not float`. I thought `bins` *is* expected to be integral? Looking at the existing `histc_cpu`,
```void THTensor_(histc)(THTensor *hist, THTensor *tensor, int64_t nbins, scalar_t minvalue, scalar_t maxvalue)```

Reply:
Resolved offline, this isn't causing any compilation errors.